vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/SelectionPreservationStrategy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2000-2024 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.shared;
+
+/**
+ * Represents selection preservation strategy on data change.
+ * <p>
+ * These enums should be used in
+ * {@link SelectionPreservationHandler#setSelectionPreservationStrategy(SelectionPreservationStrategy)}
+ * to switch between the implemented selection preservation strategies.
+ *
+ * @see SelectionPreservationHandler
+ * @author Vaadin Ltd.
+ */
+public enum SelectionPreservationStrategy {
```

Comment:
Some thoughts on API naming. While the implementation is technically a strategy pattern, we don't really use this term in our APIs, and for the developer it doesn't really matter how we have implemented it. We do use "mode" once or twice, so maybe `SelectionPreservationMode` would be a better fit, and IMO it sounds less involved than "strategy".

Reply:
Updated the name and the corresponding JavaDocs.